### PR TITLE
RavenDB-8369 changes after code review

### PR DESCRIPTION
**Pull request by myarichuk**
changes:
* code review related changes
* make sure to store last Etag at all times
* more tests

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs [hunk 1]
```diff
@@ -1203,24 +1186,29 @@ public RunningBackup OnGoingBackup(long taskId)
...
+                }
+            }
+
+            if(minLastEtag == long.MaxValue)
+                minLastEtag = 0;
+
+            using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                foreach (var collection in _database.DocumentsStorage.GetCollections(context).ToList())
```
Comment (ayende) [inline src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:1209]: Why the `ToList` here?

Comment (myarichuk) [reply src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:1209]: No reason - forgot to remove it. Will remove with next PR

File: src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs [hunk 2]
```diff
@@ -422,7 +403,9 @@ public PeriodicBackupRunner(DocumentDatabase database, ServerStore serverStore)
                             options.OperateOnTypes |= DatabaseItemType.Tombstones;
 
                         var result = CreateBackup(options, backupFilePath, startDocumentEtag, context);
-                        lastEtag = result.GetLastEtag();
+                        lastEtag = isFullBackup ?
+                            DocumentsStorage.ReadLastEtag(tx.InnerTransaction) :
```
Comment (grisha-kotler) [inline src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:407]: Why can't we rely on the last etag from smuggler?

Comment (ayende) [reply src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:407]: Using full backup, we need to get the last db wide etag, after all the current docs, tombstones, etc.

**Pull request merged**